demoscene/templatetags/safe_markdown.py

```diff
@@ -10,25 +10,28 @@
 
 cleaner = Cleaner(
```

Comment:
Additions to the whitelist here are necessary for markdown footnotes to survive Bleach's Cleaner. Without it:

1. The `div.footnotes` container would be stripped (no easy styling crutch)
2. Links and backlinks couldn't work (no fragment ids on `li` and `sup`)

Reply:
Good call to bring this up! I'm fine with allowing all `div`s with any class attribute. Ideally markdown would have a built-in sanitisation step that would kick in for arbitrary tags in the source text, while still letting through tags that are generated by our own trusted code. But I guess no-one has seen the need to implement that yet :-(

I think the most evil thing someone can do with `<div class="...">` is replicate UI elements from elsewhere in our CSS, and I think we can live with that risk.